### DISCUSSION TITLE
make format_str to work with special treatement for info label

### DISCUSSION THREAD

**Discussion by nico1080** [mattblovell/kodi_panel#17] [Ideas]
The toml file have option to concatenate text/info label in a single file with format_str.
Some info label have "special treatment" eg: artist, codec, acodec, full_codec etc.

The format_str option is not able to concatenate theses specials treatment.

Will it possible to add a syntax to allow that?

**Comment by mattblovell** [comment]
See third bullet in possible implementation ideas for Idea #16.

**Comment by mattblovell** [comment]
A distinction may be needed between "special treatments" or display callback functions that yield a string to display versus the new possibility (from Idea #20) that result with the in-progress image being manipulated (e.g., a codec image rendered).  Only those functions that return a string lend themselves to `format_str` interpolation, naturally!

**Comment by mattblovell** [comment]
Should be possible with v1.22

Note that only string callback functions (i.e., those that are only passed enough information to do pure string manipulation) can be used as part of `format_str` interpolation.  Having the drawing-capable "element" callback functions also be available would have required passing the `image` and `draw` objects to `format_InfoLabels()`, which didn't seem desirable.
